cypress_test/integration_tests/desktop/impress/comment_switching_spec.js

```diff
@@ -0,0 +1,55 @@
+/* global describe it cy require afterEach beforeEach */
+
+var helper = require('../../common/helper');
+var { addComment, addSlide, changeSlide }= require('../../common/impress_helper');
+
+describe('Comment Switching',function() {
+	var testFileName = 'comment_switching.odp';
+
+	beforeEach(function() {
+		helper.beforeAll(testFileName, 'impress');
+	});
+
+	afterEach(function() {
+		helper.afterAll(testFileName, this.currentTest.state);
+	});
+
+	it('no comment or one comment', function() {
+		cy.get('.leaflet-control-scroll-down').should('not.exist');
+		addComment(1);
+		cy.get('.leaflet-control-scroll-down').should('not.exist');
+	});
+
+	it('omit slides without comments', function() {
+		addComment(1);
+		addSlide(2);
+		addComment(1);
+		changeSlide(2,'previous');
+		cy.get('.leaflet-control-scroll-down').should('exist');
+		cy.get('.leaflet-control-scroll-down').click();
+		cy.get('#PageStatus').should('contain','Slide 3 of 3');
+	});
+
+
+	it('switch to previous or next slide when no comments in current slide',function() {
+		addSlide(1);
+		addComment(2);
+		changeSlide(1,'previous');
+		cy.get('.leaflet-control-scroll-down').should('exist');
+		cy.get('.leaflet-control-scroll-down').click();
+		cy.get('#PageStatus').should('contain','Slide 2 of 2');
+	});
+
+
+	it('slide should not change when there are multiple comments on same slide', function() {
+		addSlide(1);
+		addComment(1);
+		changeSlide(1,'previous');
+		addComment(2);
+		var scrollDownButton = cy.get('.leaflet-control-scroll-down').should('exist');
+		for (var n=0;n<2;n++) {
+			cy.get('#PageStatus').should('contain','Slide 1 of 2');
+			scrollDownButton.click();
```

Comment:
this part is a little fishy, does not matter how many times I iterate test never fails.